MechJeb2/MechJebModuleDeployableController.cs

```diff
@@ -177,5 +137,20 @@ protected enum DeployablePartState
         }
 
         protected abstract string getButtonText(DeployablePartState deployablePartState);
+        public override void OnStart(PartModule.StartState state)
+        {
+            base.OnStart(state);
+            if (HighLogic.LoadedSceneIsFlight)
+            {
+                DiscoverDeployablePartModules();
+                GameEvents.onVesselWasModified.Add(OnVesselModified);
+            }
+        }
+        public override void OnDestroy()
+        {
+            if (HighLogic.LoadedSceneIsFlight) GameEvents.onVesselWasModified.Remove(OnVesselModified);
+            base.OnDestroy();
+        }
+        public void OnVesselModified(Vessel v) => DiscoverDeployablePartModules();
```

Comment:
This should probably get pushed down into MechJebCore and ComputerModule so that all modules get a vessel modification callback.  Also, what about `onVesselStandardModification`?  It seems to be the "superclass" of all vessel change events.

Reply:
This event handler is only relevant to the DeployableController.  I don't think there's a reason for every PM to get a callback just in case they could want to do something.  Especially if thus far none of them do. 
 Pushing it into MJCore or ComputerModule will mean that MJ registers on behalf of many handlers that will just NOP, and the pattern for registering and de-registering is straightforward enough.

You could swap to onVesselStandardModification instead, looks like.  I haven't used it, but that doesn't mean a lot.

Note there's nothing stopping ComputerModule from also wanting to subscribe to that Event, even if this module does.  The same PartModule could even conceivably subscribe to the same Event multiple times.  (This happens often, but as a coding error rather than an intentional thing.)

Reply:
Implemented in 24bc1c6